### PR TITLE
[nt] Add correlation table for high correlations

### PR DESCRIPTION
**Pull request by nathaniel-mage**
# Summary

### Display a table of high correlation values
- On the column detail visualization view, add table for "Columns with high correlation" next to the correlations heatmap, for all correlation values of abs(0.5) or more.

# Tests

### Table with a highly correlated column
image

### Table with a low correlation
image

### Table with no correlation
image

cc: @johnson-mage

### CODE REVIEW

**Comment by johnson-mage** [comment]
add some padding to the "Table with no correlation" message

**Comment by johnson-mage** [comment]
get rid of the weird rounder top borders in the subheader

**Comment by nathaniel-mage** [comment]
> get rid of the weird rounded top borders in the subheader

Can you add a screenshot of this? Not sure where this is

**Comment by johnson-mage** [comment]
> > get rid of the weird rounded top borders in the subheader
> 
> Can you add a screenshot of this? Not sure where this is
image

**Comment by johnson-mage** [comment]
the rows should have alternating background colors in the high correlations table

**Comment by nathaniel-mage** [comment]
I'm still unable to see what you mean by the borders. The screenshot didn't help because it was so zoomed in. Is this happening for all chart components thing or only on the high correlations chart?

image

I updated the spacing, and if there's anything that's chart specific I can address that in another PR to keep this clean.

**Comment by nathaniel-mage** [comment]
I'll add the table styling changes here too, but do I need to update the PR title?

**Comment by johnson-mage** [comment]
> I'll add the table styling changes here too, but do I need to update the PR title?

no

**Review by johnson-mage — APPROVED** [review]
approved after addressing comments

File: mage_ai/frontend/components/datasets/Insights/ColumnAnalysis.tsx [hunk 1]
```diff
@@ -601,7 +613,51 @@ function ColumnAnalysis({
               />
             </ChartContainer>
           }
-          right={<div />}
+          right={
+            <ChartContainer
+              noPadding={!!correlationsRowData}
+              title="Values with high correlation"
+            >
+              { highCorrelations.length > 0 ? (
```
Comment (tommydangerous) [inline mage_ai/frontend/components/datasets/Insights/ColumnAnalysis.tsx:621]: Why do u have a space between `{` and `highCorrelations`?

File: mage_ai/frontend/components/datasets/Insights/ColumnAnalysis.tsx [hunk 2]
```diff
@@ -138,11 +138,23 @@ function ColumnAnalysis({
...
+  const highCorrelations = [];
   if (correlationsRowData) {
     correlationsRowData?.map(([, col, r], idx: number) => {
       yLabels.push(col);
       heatmapData.push([roundNumber(r)]);
     });
+    correlationsRowData?.map((_, col, r) => {
+      if (Math.abs(r[col][2]) > 0.5) {
+        highCorrelations.push({
+          'columnValues': [
```
Comment (tommydangerous) [inline mage_ai/frontend/components/datasets/Insights/ColumnAnalysis.tsx:150]: remove single quotes

File: mage_ai/frontend/components/datasets/Insights/ColumnAnalysis.tsx [hunk 3]
```diff
@@ -601,7 +613,51 @@ function ColumnAnalysis({
...
+                  ]}
+                  noBorder
+                  rowGroupData={[
+                    {
+                      rowData: highCorrelations,
+                    },
+                  ]}
+                  small
+                />
+              ) : (
```
Comment (tommydangerous) [inline mage_ai/frontend/components/datasets/Insights/ColumnAnalysis.tsx:643]: Multi-line ternary operator style is like this:

```
true_or_false
  ? something
  : something
```

Comment (nathaniel-mage) [reply mage_ai/frontend/components/datasets/Insights/ColumnAnalysis.tsx:643]: got it, will update this in #250

File: mage_ai/frontend/oracle/components/Table/index.style.tsx [hunk 4]
```diff
@@ -40,6 +39,10 @@ export const ColumnHeaderRowStyle = styled.div<any>`
   ${props => !props.noBorder && `
     border: 1px solid ${(props.theme.interative || light.interactive).defaultBorder};
   `}
+
+  ${props => props.noBorder && `
+    border-radius: 0px 0px 0px 0px;
```
[the remark this answers is not in the document]
Comment (nathaniel-mage) [reply mage_ai/frontend/oracle/components/Table/index.style.tsx:44]: Before when I was testing I was looking at different corners to see why there was a faded border. I'll update it to say 0 in #250

**Comment by nathaniel-mage** [comment]
For these remaining bugs I'll open a new PR and do it on a fresh branch instead.

**Pull request merged**